src/Symfony/Component/Config/ResourceCheckerConfigCache.php

```diff
@@ -156,7 +156,11 @@ private function safelyUnserialize(string $file)
         });
 
         try {
-            $meta = unserialize($content);
+            try {
+                $meta = unserialize($content);
+            } catch (\UnserializationFailedException $e) {
+                throw $e instanceof \UnserializationFailedException && $e->getPrevious() === $signalingException ? $signalingException : $e;
```

Comment:
`safeUnserialize()` registers both an error handler and an "unserialize_callback_func" (see L149) to turn any PHP warning/notice/etc. + class-not-founds into `$signalingException`. Unwrapping `UnserializationFailedException` is required to work around the RFC.